Makefile

```diff
@@ -332,6 +349,8 @@ precommit: prerequisites
 	$(Q)+$(MAKE) --no-print-directory clean
 	$(Q) echo "Running checks..."
 	$(Q)+$(MAKE) --no-print-directory check-signed-off check-vera check-cppcheck
+	$(Q) echo "...testing build-options..."
+	$(Q)+$(MAKE) --no-print-directory test-buildoptions
```

Comment:
I'm concerned about performance here. I quite often just run `make` (== `make precommit`) to see that everything is all right (checks, builds, tests). However, with this addition, this will perform all the builds with various options too. And this will call out to a shell script, which sequentially walks through targets and options, and then calls back to make again, thus loosing all parallelization potential. (Not to mention that ninja builds are way faster than non-ninja builds, but even if I run `make precommit NINJA=1`, because of the make->shell->make calls, this setting will be lost and loose ninja as well.) Is there no way to get everything done within the Makefile?